File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.py
class Solution:
    def subarraySum(self, nums, k):

        hm = {0:1}
        prefix_sum = 0
        count = 0

        for num in nums:

            prefix_sum += num
            diff = prefix_sum - k

            if diff in hm:
                count += hm[diff]

            hm[prefix_sum] = hm.get(prefix_sum,0) + 1

        return count

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.java
class Solution {
    public int longestConsecutive(int[] nums) {
        
		Set<Integer> numSet = new HashSet<>();
		for (int num : nums) {
			numSet.add(num);
		}

		int longest = 0;

//		100 4 200 1 3 2
		for (int n : numSet) {
			int length = 1;
			while (numSet.contains(n + length)) {
				length++;
			}
			longest = Math.max(length, longest);
		}
		return longest;
	
    }
}
